cheatsheets/Secure_Product_Design_Cheat_sheet.md

```diff
@@ -0,0 +1,92 @@
+
+Adding (or removing) connections is probably a sign that **_Product Inception_** is happening.
+
+### 4. Code
+
+Code is the ultimate expression of the intention for a product and as such it must be functional first and foremost. But there is a quality to how that functionality is provided that must meet or exceed the expectations of it.
+
+Some basics of secure coding include:
+
+   1. Input validation: Verify that all input data is valid and of the expected type, format, and length before processing it. This can help prevent attacks such as SQL injection and buffer overflow.
+   2. Error handling: Handle errors and exceptions in a secure manner, such as by logging them in a secure way and not disclosing sensitive information to an attacker.
+   3. Authentication and Authorization: Implementing strong authentication and authorization mechanisms to ensure that only authorized users can access sensitive data and resources.
+   4. Cryptography: Use cryptographic functions and protocols to protect data in transit and at rest, such as HTTPS and encryption - the expected levels for a given Product Security Level can often be found by reviewing your Golden Path / Paved Road documentation.
+   5. Least privilege: Use the principle of the least privilege when writing code, such that the code and the system it runs on are given the minimum access rights necessary to perform their functions.
+   6. Secure memory management: Use high-level languages recommended in your Golden Path / Paved Road documentation or properly manage memory to prevent memory-related vulnerabilities such as buffer overflow and use-after-free.
+   7. Avoiding hardcoded secrets: Hardcoded secrets such as passwords and encryption keys should be avoided in the code and should be stored in a secure storage.
+   8. Security testing: Test the software for security vulnerabilities during development and before deployment.
+   9. Auditing and reviewing the code: Regularly audit and review the code for security vulnerabilities, such as by using automated tools or having a third party review the code.
+   10. Keeping up-to-date: Keep up-to-date with the latest security best practices and vulnerabilities to ensure that the software is as secure as possible.
+
+Ensure that you integrate plausibility checks at each tier of your application (from frontend to backend) and ensure that you write unit and integration tests to validate that all critical flows are resistant to the results of [Threat Modelling](Threat_Modeling_Cheat_Sheet.md). Use that to compile use-cases and misuse-cases for each tier of your application.
```

Comment:
```suggestion
Ensure that you integrate plausibility checks at each tier of your application (e.g., from frontend to backend) and ensure that you write unit and integration tests to validate that all critical flows are resistant to the results of your [Threat Modelling](Threat_Modeling_Cheat_Sheet.md). Use that to compile use-cases and misuse-cases for each tier of your application.
```

Reply:
Hmm... I'm not fond of the phrasing "all critical flows are resistant to the results of your Threat Modelling".

Maybe a better way of stating that would be something like "all threats discovered during Threat Modelling have been mitigated to a level of risk acceptable to the organization"???